### DISCUSSION TITLE
Arcjet rate limiting algorithms

### DISCUSSION THREAD

**Discussion by giscus** [arcjet/arcjet-docs#203] [General]
# Arcjet rate limiting algorithms

Arcjet rate limiting supports fixed window, sliding window, and token bucket algorithms.

https://docs.arcjet.com/rate-limiting/algorithms/

**Comment by gabrielgpavao** [comment]
Please, I am sorry if I got it wrong, but the math seems to be wrong at line 5 of the second code block inside the Sliding Window section.
50 * ((60 - 30) / 60) + 50 **is not equal** to 100

50 * ((60 - 30) / 60) + 50
50 * (30 / 60) + 50
50 * 0,5 + 50
25 + 50
= 75

So, in that case, the client would not be blocked from making further requests, for it still has remaining amount of 25 requests.

remaining = limit - rate = 0
remaining = 100 - 75 = 25

**Reply by davidmytton** [reply]
Thanks, @gabrielgpavao - you're right! I've corrected the example in https://github.com/arcjet/arcjet-docs/pull/204
